controllers/performanceprofile_controller.go

```diff
@@ -140,14 +141,14 @@ func (r *PerformanceProfileReconciler) mcpToPerformanceProfile(mcpObj client.Obj
 
 	var requests []reconcile.Request
 	for i, profile := range profiles.Items {
-		machineConfigPoolSelector := labels.Set(profileutil.GetMachineConfigPoolSelector(&profile))
-		selector, err := metav1.LabelSelectorAsSelector(mcp.Spec.MachineConfigSelector)
+		profileNodeSelector := labels.Set(profile.Spec.NodeSelector)
```

Comment:
I'm not completely clear why we need this logic (this is a part of the code I don't really grok). Could you please elaborate to help me understand?

Reply:
I will start with the method description, we are using this method to initiate reconcile loop for MCP objects. But the reconcile loop should work only on one type of object, in our case, it is the performance profile. So during MCP create, update or delete our watch logic get the MCP object and trying to fetch from it the relevant performance profile, to fetch it the code checks the MCP node selector and finds all performance profiles with the same node selector. I hope it makes things clearer, give me know if you need a more descriptive explanation.